### ISSUE TITLE
Make /proc/<pid>/root available to other users

### ISSUE THREAD

**Issue by hadess** [flatpak/flatpak#1644]
geoclue currently uses a similar technique that the portals do to detect whether an application is sandboxed. This however means that geoclue isn't able to detect whether applications are sandboxed when run as the `geoclue` user instead of root.

Making `/proc/<pid>/root` available to other users would make it possible for geoclue to determine whether an application is sandboxed, without running as root.

See https://bugs.freedesktop.org/show_bug.cgi?id=97776#c16

**Comment by matthiasclasen** [comment]
Imo the geoclue portal stuff needs to be redone as a session service.

**Comment by hadess** [comment]
There's no need for that, there's already a session component to the process, in gnome-shell, and using it to determine the Flatpak ID is what I proposed in https://bugs.freedesktop.org/show_bug.cgi?id=97776#c22 if this problem cannot be solved.

**Comment by matthiasclasen** [comment]
The fact that you need flatpak changes shows that there is a need for that. 
The way the geoclue stuff is done, with the session component registering with the system service, is really awkward and not at all how I think this should work.

**Comment by hadess** [comment]
You keep saying that, and I keep having to explain why Geoclue can't run in the session, or just be a Portal.

**Comment by alexlarsson** [comment]
We can't just "make it available". Its inaccessible to other users for complex kernel reasons that we don't really control.

**Comment by matthiasclasen** [comment]
Geoclue itself maybe can't run in the session. But that does not mean that we can't have a location portal

**Comment by alexlarsson** [comment]
Yeah, i agree. The whole thing `app -> system service -> back to some thing in the user session` thing is really weird. It breaks when there are multiple user sessions, and it generally seems wrong for a highly privileged service to talk to something per-user. I think a portal in the user session that acts as a verifying proxy for the system service is a much nicer design.
